kedro-datasets/setup.py

```diff
@@ -18,14 +18,17 @@ def _collect_requirements(requires):
 api_require = {"api.APIDataSet": ["requests~=2.20"]}
 biosequence_require = {"biosequence.BioSequenceDataSet": ["biopython~=1.73"]}
 dask_require = {
-    "dask.ParquetDataSet": ["dask[complete]>=2021.10", "triad>=0.6.7, <1.0"]
+    "dask.ParquetDataSet": ["dask[complete]>=2021.12.0", "triad>=0.6.7, <1.0"]
```

Comment:
Why is this needed?

Reply:
Related to https://github.com/kedro-org/kedro/issues/3178 ? 🤔

Reply:
Actually looks like a larger issue here: https://github.com/kedro-org/kedro/issues/3178

Reply:
This bump of the pinned upper bound shouldn't be needed. If a user is only using the `dask` dataset or only the `netncfdataset` they should be able to without any dependency issues. If they happen to have both installed, they just need to bump the dependencies however needed. It was pinned to `>=2021.10` so that means that if the `xarray` installed needs a newer version that is possible without having to pin `dask` to a newer version in `kedro-datasets`.

Reply:
Yes that makes a lot of sense in theory! I'll bump it back down, and see what happens. The tests were breaking at this point and only resolved when locking the version higher.